equinix/resource_fabric_cloud_router.go

```diff
@@ -15,47 +18,250 @@ import (
 
 	v4 "github.com/equinix-labs/fabric-go/fabric/v4"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func resourceCloudRouter() *schema.Resource {
+func FabricCloudRouterPackageSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"code": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Fabric Cloud Router package code",
+		},
+	}
+}
+func FabricCloudRouterAccountSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"account_number": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Optional:    true,
+			Description: "Account Number",
+		},
+	}
+}
+func FabricCloudRouterProjectSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
+			Description: "Project Id",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Unique Resource URL",
+		},
+	}
+}
+
+func FabricCloudRouterResourceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Equinix-assigned Fabric Cloud Router identifier",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Fabric Cloud Router URI information",
+		},
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringLenBetween(1, 24),
+			Description:  "Fabric Cloud Router name. An alpha-numeric 24 characters string which can include only hyphens and underscores",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Customer-provided Fabric Cloud Router description",
+		},
+		"state": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Fabric Cloud Router overall state",
+		},
+		"equinix_asn": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Equinix ASN",
+		},
+		"package": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			Description: "Fabric Cloud Router Package Type",
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: FabricCloudRouterPackageSch(),
+			},
+		},
+		"change_log": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Captures Fabric Cloud Router lifecycle change information",
+			Elem: &schema.Resource{
+				Schema: equinix_fabric_schema.ChangeLogSch(),
+			},
+		},
+		"type": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice([]string{"XF_ROUTER"}, true),
+			Description:  "Defines the FCR type like XF_ROUTER",
```

Comment:
This should be updated to a more professional description. `Defines the FCR type; i.e. XF_ROUTER`